src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/ShowDropColumnsBase.cs

```diff
@@ -0,0 +1,222 @@
+
+            if (fArgsValid)
+            {
+                fArgsValid = base.CheckType(context, args[0], argTypes[0], isRecord ? DType.EmptyRecord : ParamTypes[0], errors, ref nodeToCoercedTypeMap);
+            }
+
+            var supportColumnNamesAsIdentifiers = context.Features.SupportColumnNamesAsIdentifiers;
+            var colsToKeep =
+                _isShowColumns
+                ? (isRecord ? DType.EmptyRecord : DType.EmptyTable)
+                : returnType;
+
+            var count = args.Length;
+            for (var i = 1; i < count; i++)
+            {
+                string expectedColumnName = null;
+                var nameArg = args[i];
+                var nameArgType = argTypes[i];
+
+                // Verify we have a string literal for the column name. Accd to spec, we don't support
+                // arbitrary expressions that evaluate to string values, because these values contribute to
+                // type analysis, so they need to be known upfront (before DropColumns executes).                            
+                if (supportColumnNamesAsIdentifiers)
+                {
+                    if (nameArg is not FirstNameNode identifierNode)
+                    {
+                        fArgsValid = false;
+
+                        // Argument '{0}' is invalid, expected an identifier.
+                        errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrExpectedIdentifierArg_Name, nameArg.ToString());
+                        continue;
+                    }
+
+                    expectedColumnName = identifierNode.Ident.Name;
+                }
+                else
+                {
+                    StrLitNode strLitNode = nameArg.AsStrLit();
+
+                    if (nameArgType.Kind != DKind.String || strLitNode == null)
+                    {
+                        fArgsValid = false;
+
+                        // Argument '{0}' is invalid, expected a text literal.
+                        errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrExpectedStringLiteralArg_Name, nameArg.ToString());
+                        continue;
+                    }
+
+                    expectedColumnName = strLitNode.Value;
+                }
+
+                // Verify that the name is valid.
+                if (!DName.IsValidDName(expectedColumnName))
+                {
+                    fArgsValid = false;
+
+                    // Argument '{0}' is not a valid identifier.
+                    errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrArgNotAValidIdentifier_Name, expectedColumnName);
+                    continue;
+                }
+
+                var columnName = _isShowColumns ?
+                    new DName(expectedColumnName) :
+                    new DName(
+                        DType.TryGetLogicalNameForColumn(argTypes[0], expectedColumnName, out var logicalName)
```

Comment:
>DType.TryGetLogicalNameForColumn(argTypes[0], expectedColumnName, out var logicalName) [](http://example.com/codeflow?start=24&length=86)

Display names should work for both `DropColumns` and `ShowColumns`.

Reply:
Updated